apps/backoffice/src/components/services/subscriptions-migration/migration-latest.tsx

```diff
@@ -0,0 +1,111 @@
+import { LoaderSkeleton } from "@/components/loaders";
+import { MigrationItemList } from "@/generated/api/MigrationItemList";
+import { MigrationItemStatus } from "@/generated/api/MigrationItemStatus";
+import { Refresh } from "@mui/icons-material";
+import {
+  Alert,
+  Box,
+  Button,
+  Chip,
+  Grid,
+  Stack,
+  Typography
+} from "@mui/material";
+import { useTranslation } from "next-i18next";
+import { Fragment, useEffect, useState } from "react";
+import { MigrationChipStatus } from ".";
+
+export type MigrationLatestProps = {
+  migrationItems?: MigrationItemList;
+  onRefreshClick: () => void;
+};
+
+/** Render subscriptions migration latest summary status */
+export const MigrationLatest = ({
+  migrationItems,
+  onRefreshClick
+}: MigrationLatestProps) => {
+  const { t } = useTranslation();
+  const [isEmptyState, setIsEmptyState] = useState(false);
+
+  // the full migration status report is reduced so that it's ready to be rendered
+  const computeMigrationStatus = (
+    status: MigrationItemStatus
+  ): MigrationChipStatus => {
+    // at least one failed mean the overall process failed
+    return status.failed > 0
+      ? { label: "failed", color: "error" }
+      : // at least one still processing means the overall process still processing
+      status.processing > 0
+      ? { label: "doing", color: "warning" }
+      : // for every other case, we consider it done
+        { label: "done", color: "success" };
+  };
+
+  useEffect(() => {
+    setIsEmptyState(migrationItems?.items?.length === 0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [migrationItems]);
+
```

Comment:
Wouldn't this work in the same way? (instead of using state)

```suggestion
const isEmptyState = (migrationItems?.items?.length === 0)
```

Reply:
it's not a good practice to do a direct assignment that way, especially if the value changes dynamically during renders

Reply:
on the contrary, the best practice is to avoid `useEffect` whenever is it possible, expecially with derived properties. 

https://stackoverflow.com/questions/68166155/consequences-of-using-computed-variables-vs-usestate-useeffect

> The only reason to use hooks or other state tracking functionality is if you plan on changing the state within the component itself.

Reply:
see also https://prateeksurana.me/blog/why-you-should-avoid-using-state-for-computed-properties/

Reply:
1f277d8
you're absolutely right and I think I've always misunderstood this concept, thank you very much, I won't forget it!